include/deal.II/matrix_free/operators.h

```diff
@@ -2382,7 +2382,8 @@ namespace MatrixFreeOperators
         VectorizedArrayType local_diagonal_vector[phi.static_dofs_per_cell];
         for (unsigned int i = 0; i < phi.dofs_per_component; ++i)
           {
-            for (unsigned int j = 0; j < phi.dofs_per_component; ++j)
+            for (unsigned int j = 0; j < phi.dofs_per_component * n_components;
```

Comment:
What about `i`?

Reply:
We do not need that one because the code only computes the first component of a system and then duplicates entries: https://github.com/dealii/dealii/blob/1861065e3a13a4b12320a43f28a127ed142fdcf4/include/deal.II/matrix_free/operators.h#L2391-L2394

Reply:
Thanks for the explanation. Makes sense. We are computing a rectangle here :+1:

Reply:
So the bug was that we did not fill the second and third component initially, so we could get unitialized data there.

Reply:
In theory, we could zero out component 2 and 3 once at the beginning and don't alter it later on? The result is block diagonal, isn't it?

Reply:
Yes that should be possible, as we explicitly assume that components do not mix.